ecs-init/cache/cache_test.go

```diff
@@ -332,10 +354,13 @@ func TestDownloadAgentSuccess(t *testing.T) {
 	})
 	mockFS.EXPECT().Copy(tempfile, gomock.Any()).Return(int64(0), nil)
 	mockFS.EXPECT().Rename(tempfile.Name(), config.AgentTarball())
+	mockMetadata := NewMockinstanceMetadata(mockCtrl)
+	mockMetadata.EXPECT().Region().Return(config.DefaultRegionName, nil)
```

Comment:
Same as above. Can we be more strict and the check the ordering here (Using [`gomock.InOrder`](https://godoc.org/github.com/golang/mock/gomock#InOrder))?